app/controllers/tools_controller.rb

```diff
@@ -1,77 +1,32 @@
 class ToolsController < ApplicationController
   require 'saml_idp'
 
-  def index
-    flash[:error] = nil
+  def saml_request
+    flash[:warning] = nil
+    @validation_attempted = true
 
-    if !auth_url.present?
-      flash[:error] = 'Please submit an auth URL or SAMLRequest to be validated.'
+    if params['validation'].blank?
+      @validation_attempted = false
       return
     end
 
-    begin
-      certs = (cert_param || sp.certs).map { |cert| OpenSSL::X509::Certificate.new(cert) }
-    rescue OpenSSL::X509::CertificateError
-      flash[:error] = 'Something is wrong with the certificate you submitted.'
-    rescue NoMethodError
-      flash[:error] = 'Could not find any certificates to use. Please add a certificate to your application configuration or paste one below.'
-    end
-
-    auth_service_provider&.certs = certs
-    @valid_request = valid_request
-
-    @valid_signature = valid_signature
-    @matching_cert_sn = matching_cert_sn
+    @request = Tools::SAMLRequest.new(validation_params)
 
-    if auth_request && @valid_request
-      xml = REXML::Document.new(auth_request.raw_xml)
-      xml.write(@xml = '', 2)
+    if @request.logout_request?
+      flash[:warning] = 'You have passed a logout request. Currently, this tool is for ' +
+                        'Authentication requests only.'
```

Comment:
Do we want to add a link somewhere to the SAML tool for testing logout requests for now until we have our own tool?

Reply:
yes!

Reply:
thanks!